### PR TITLE
Add error [Errno 104] Connection reset by peer

### LINKED ISSUE #133: error [Errno 104] Connection reset by peer
### Discussed in https://github.com/oppia/oppia/discussions/17596

<div type='discussions-op-text'>

<sup>Originally posted by **Vansh-31** March  6, 2023</sup>
### Operating System

Ubuntu 22.04

### RAM size

6 GB

### Which step are you stuck on?

python -m scripts.start

### Error Log

```bash
Downloading file ui-leaflet.min.no-header.js to ./third_party/static/angular-ui-leaflet-1.0.3 ...
Attempt 1 of 2 failed when downloading https://raw.githubusercontent.com/angular-ui/ui-leaflet/v1.0.3/dist/ui-leaflet.min.no-header.js.
Error: <urlopen error [Errno 104] Connection reset by peer>
Retrying download.
Attempt 2 of 2 failed when downloading https://raw.githubusercontent.com/angular-ui/ui-leaflet/v1.0.3/dist/ui-leaflet.min.no-header.js.
Traceback (most recent call last):
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 1354, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/http/client.py", line 1256, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/http/client.py", line 1302, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/http/client.py", line 1251, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/http/client.py", line 1011, in _send_output
    self.send(msg)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/http/client.py", line 951, in send
    self.connect()
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/http/client.py", line 1425, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/vansh/Desktop/Open Source/oppia/scripts/start.py", line 32, in <module>
    install_third_party_libs.main()
  File "/home/vansh/Desktop/Open Source/oppia/scripts/install_third_party_libs.py", line 172, in main
    install_third_party.main(args=[])
  File "/home/vansh/Desktop/Open Source/oppia/scripts/install_third_party.py", line 522, in main
    download_all_dependencies(DEPENDENCIES_FILE_PATH)
  File "/home/vansh/Desktop/Open Source/oppia/scripts/install_third_party.py", line 376, in download_all_dependencies
    download_files(dependency_url, dependency_dst, dependency_files)
  File "/home/vansh/Desktop/Open Source/oppia/scripts/install_third_party.py", line 145, in download_files
    common.url_retrieve(
  File "/home/vansh/Desktop/Open Source/oppia/scripts/common.py", line 876, in url_retrieve
    raise exception
  File "/home/vansh/Desktop/Open Source/oppia/scripts/common.py", line 864, in url_retrieve
    with urlrequest.urlopen(
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 1397, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/urllib/request.py", line 1357, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 104] Connection reset by peer>
vansh@Vanshs-device:~/Desktop/Open Source/oppia$
```

### Approaches already used to resolve the issue

tried python -m scripts.clean
disabled firewall

### Additional information

_No response_</div>

## Solution
Run
```
pip install "requests[security]"
```

### PR DESCRIPTION
**Pull request by deadex-ng**
This fixes #133 
It also arranges the Table of Contents in alphabetical order to easily navigate through.

### CODE REVIEW

**Review by U8NWXD — CHANGES_REQUESTED** [review]
The table of contents is out of order compared to the rest of the document

File: Troubleshooting.md [hunk 1]
```diff
@@ -3,53 +3,83 @@
...
+    self.sock = self._context.wrap_socket(self.sock,
+  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/ssl.py", line 500, in wrap_socket
+    return self.sslsocket_class._create(
+  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/ssl.py", line 1040, in _create
+    self.do_handshake()
+  File "/home/vansh/.pyenv/versions/3.8.15/lib/python3.8/ssl.py", line 1309, in do_handshake
+    self._sslobj.do_handshake()
+ConnectionResetError: [Errno 104] Connection reset by peer
+```
+Then run `pip install "requests[security]"`.
```
Comment (gp201) [inline Troubleshooting.md]:
```suggestion
Then run `pip install requests[security]`. 
```

**Review by gp201 — COMMENTED** [review]
1 minor fix

File: Troubleshooting.md (hunk 1 above)
Comment (deadex-ng) [reply Troubleshooting.md]: I've updated this in the PR.

**Review by U8NWXD — APPROVED** [review]

**Pull request merged**